source/daplink/interface/swd_host.c

```diff
@@ -156,22 +156,16 @@ uint8_t swd_write_dp(uint8_t adr, uint32_t val)
     uint8_t data[4];
     uint8_t ack;
 
-    switch (adr) {
-        case DP_SELECT:
-            if (dap_state.select == val) {
-                return 1;
-            }
-
-            dap_state.select = val;
-            break;
-
-        default:
-            break;
-    }
+    //check if the right bank is already selected
+    if(adr == DP_SELECT && dap_state.select == val)
```

Comment:
Please check formatting on these changes: Spaces between **if** and other keywords and parentheses, and between parentheses and braces. No spaces on inside of parentheses, eg `(foo)` not `( foo )`. Always use braces even for single statements (like line 161 here). Sorry to be pedantic 😁